File: src/main/java/random/telegramhomebot/scheduling/PingStoredHostsScheduler.java
package random.telegramhomebot.scheduling;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.annotation.Profile;
import org.springframework.scheduling.annotation.Async;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Service;
import random.telegramhomebot.config.ProfileService;
import random.telegramhomebot.services.HostService;

@Slf4j
@RequiredArgsConstructor
@Profile(ProfileService.NETWORK_MONITOR)
@Service
public class PingStoredHostsScheduler {

	private final HostService hostService;

	@Async
	@Scheduled(fixedRateString = "${ping.stored.hosts.scheduled.time}", initialDelay = 10000)
	public void pingStoredHosts() {
		log.debug("PingStoredHostsScheduler :: Ping stored hosts...");
		hostService.pingStoredHosts();
	}
}

File: src/main/java/random/telegramhomebot/scheduling/PingStoredHostsScheduler.kt
package random.telegramhomebot.scheduling

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.scheduling.annotation.Async
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Service
import random.telegramhomebot.services.hosts.HostService
import random.telegramhomebot.utils.logger

@ConditionalOnProperty(prefix = "network-monitor", value = ["enabled"], havingValue = "true")
@Service
class PingStoredHostsScheduler(private val hostService: HostService) {
    val log = logger()

    @Async
    @Scheduled(fixedRateString = "\${ping.stored.hosts.scheduled.time}", initialDelay = 10000)
    fun pingStoredHosts() {
        log.debug("PingStoredHostsScheduler :: Ping stored hosts...")
        hostService.pingStoredHosts()
    }
}
